### PR TITLE
Leader election: clean up and add leader election to agent

### PR DESCRIPTION
**Pull request by Garrybest**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Now the leader election is a mess, some module uses leader election as default options but some other does not. So I try to clean up this leader election code. Meanwhile, `karmada-agent` now does not use leader election so if we deploy 2 replicas, there will arise some competition errrors.

Now leader election will be a default option. The lease will be created into `karmada-system`.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Comment by RainbowMango** [comment]
My bad. Can't all components work in leader election mode?
I remember I've tested the `karmada-controller-manager` at a very early stage.

Maybe we can set replicas to 3 at [default artifacts](https://github.com/karmada-io/karmada/tree/master/artifacts), then we can test it by CI.

**Comment by Garrybest** [comment]
The `karmada-controller-manager` is ok when using leader election mode, just need some cleanup, e.g., move lease into `karmada-system` namespace instead of `default`. 

The `karmada-scheduler` does not create leader election. When multi-replicas have been created, cluster health lease will be updated by two replicas which lead to some errrors.

https://github.com/karmada-io/karmada/blob/38f16d63b74d7a5d840ff8e7cdfac660e9fe8d0e/cmd/agent/app/agent.go#L69-L75

The `karmada-scheduler` does not run within leader election mode by default.

https://github.com/karmada-io/karmada/blob/38f16d63b74d7a5d840ff8e7cdfac660e9fe8d0e/cmd/scheduler/app/options/options.go#L56-L56

There is no need for the `karmada-webhook` to use leader election, it is just a HTTP router. Code also needs clean up when leader election is not necessary.

File: cmd/controller-manager/app/options/options.go [hunk 1]
```diff
@@ -57,34 +51,13 @@ type Options struct {
...
-		o.LeaderElection.RenewDeadline = defaultElectionRenewDeadline
-		klog.Infof("Set default value: Options.LeaderElection.RenewDeadline = %s", defaultElectionRenewDeadline.Duration.String())
-	}
-
-	if o.LeaderElection.RetryPeriod.Duration.Seconds() == 0 {
-		o.LeaderElection.RetryPeriod = defaultElectionRetryPeriod
-		klog.Infof("Set default value: Options.LeaderElection.RetryPeriod = %s", defaultElectionRetryPeriod.Duration.String())
+	return &Options{
+		LeaderElection: componentbaseconfig.LeaderElectionConfiguration{
+			LeaderElect:       false,
```
[the remark this answers is not in the document]
Comment (Garrybest) [reply cmd/controller-manager/app/options/options.go:56]: > What's the problem with the `Complete()`?
1. The variable initialized in this function has not been used in any other place. So it is redundant.
2.  To keep the code concise, we don't have to set `leaseDuration`, `renewDeadline`, and `retryPeriod` by default becase they are set in `controller-runtime` with the same value.
https://github.com/karmada-io/karmada/blob/b9b0c0b965fcd5ab2bc12a11ee28df050a441a94/vendor/sigs.k8s.io/controller-runtime/pkg/manager/manager.go#L498-L502

> Here the LeaderElect default value is false but we have set it's default with true.
Well, it doesn't matter becase `AddFlags` will set this value finally. But keeping them consistent is a good idea.

[the remark this answers is not in the document]
Comment (RainbowMango) [reply cmd/controller-manager/app/options/options.go:56]: > But keeping them consistent is a good idea.

Yes, that's my concern.

**Comment by RainbowMango** [comment]
LGTM.
Just want to confirm if the behavior is like:
| component | support leader election | default leader election | support multi-replicas |
| ---- | ---- | ---- | ---- |
| `karmada-controller-manager` | true | true | true |
| `karmada-agent` | true | true | true |
| `karmada-scheduler` | true | true | true |
| `karmada-webhook` | false | false | true |

**Comment by Garrybest** [comment]
> LGTM.
> Just want to confirm if the behavior is like:
> 
> component	support leader election	default leader election	support multi-replicas
> `karmada-controller-manager`	true	true	true
> `karmada-agent`	true	true	true
> `karmada-scheduler`	true	true	true
> `karmada-webhook`	false	false	true

That's right.

**Pull request merged**